docs/pages/blog/turbo-1-4-0.mdx

```diff
@@ -0,0 +1,134 @@
+
+And, since the work for your build is the driving factor of how much work needs to be done, we can’t cut corners there.
+
+### Do the work when nobody is watching.
+
+So, if we can’t make it faster and we can’t do less work, the best way we can make things _feel_ faster is to do the work when nobody is watching. What if Turborepo could do all of the work that you care about right now and then handle all of the bookkeeping later?
+
+## Agents for change
+
+Now that we have the Turborepo Agent, we’re excited to be able to continue improving your build experience in the future. There are two major categories of performance-improvement opportunities that we see it enabling:
+
+- Filesystem monitoring
+- Cache management
+
+### Filesystem monitoring
+
+Right now, the Turborepo Agent has been given just one job: monitor the output locations of tasks. But having filesystem monitoring in place enables additional opportunities with tracking of task inputs to precompute task hashes and perform differential cache restoration.
+
+### Cache management
+
+The last step of any Turborepo build is to store the build outputs with the remote cache. Depending on your network connection this can be either near-instantaneous or prohibitively slow. We plan to hand that job off to the Turborepo Agent so you’ll never have to wait for network calls again.
+
+## Under the hood of Turborepo
+
+Turborepo Agent has been a major effort including decisions about the future architecture of Turborepo.
+
+[![Header of PR which landed the Turborepo Agent.](/images/blog/turbo-1-4-0/pr.png)](https://github.com/vercel/turborepo/pull/1429)
+
+Now, when you start up `turbo`, the first thing we do is launch the Turborepo Agent. We have to wait for that to happen so we use a binary exponential backoff for our connection attempts. This ensures we get out of the way as quickly as possible so you can start building.
+
+That interprocess communication uses gRPC and protobufs to give us versioning guarantees (both backwards- and forwards-compatible) and quick marshaling of data so that we don’t spend all of our time chatting.
+
+Unsupervised processes are wasteful and the majority of the code that was added was to make process handling as robust as possible.
+
+## New examples
+
+Over the past month we’ve invested a lot of time in [creating new examples for how to integrate Turborepo into your workflow](https://github.com/vercel/turborepo/tree/main/examples), featuring:
+
+- Svelte
+- Docker
+- Create React App
+- React Native
+- Prisma
+- Remix
+- … and more!
+
+## Automatic environment variable inclusion
+
+One of the easiest mistakes to make when setting up Turborepo is accidentally omitting an important environment variable resulting in reusing a cache somewhere you didn't expect. Turborepo now has your back and will detect commonly used environment variables to help you avoid this problem.
```

Comment:
Maybe worth listing the frameworks here that we auto detect?

Reply:
Enumerated.